File: src/main/webapp/js/template/data-statistics/dataStatisticsTemplate.js
/**
 * Created by zhtt on 2016/8/8.
 */
$(function(){
    dataStatisticsTemplate.init();
});
var dataStatisticsTemplate={
    getUrl:ctx+"/template/data-statistics/get/",
    moveUrl:ctx+"/template/data-statistics/move/",
    changeStateUrl:ctx+"/template/data-statistics/show/",
    deleteUrl:ctx+"/template/data-statistics/delete/",
    tableUrl:ctx+"/template/data-statistics/table?table=false",
    treeUrl:ctx+"/template/data-statistics/tree?tree=false",
    treeId:"dataStatisticsTemplateTreeDiv",
    createRootModalId:"openCreateRootDataStatisticsTemplateModelBtn",
    createGroupModalId:"openCreateGroupDataStatisticsTemplateModelBtn",
    createFieldModalId:"openCreateFieldDataStatisticsTemplateModelBtn",
    createUrl:ctx+"/template/data-statistics/form",
    cloneTreeUrl:ctx+"/template/data-statistics/clone-tree",
    tree:null
};
dataStatisticsTemplate.init=function(){
    dataStatisticsTemplateTree.init();
    dataStatisticsTemplateTable.init();
    dataStatisticsTemplate.initEvent();
}
dataStatisticsTemplate.initEvent=function(){
    $('#createGroupDataStatisticsTemplateModal').on('shown.bs.modal', function () {
        $("#updateRootDataStatisticsTemplateModal .crete-close-btn").show();
        $("#updateRootDataStatisticsTemplateModal .update-close-btn").hide();
        var selected=dataStatisticsTemplate.tree.getSelectedNodes();
        $('#createGroupDataStatisticsTemplateModal input[name="parentId"]').val(selected.nodes[0].uuid);
        $("#createGroupDataStatisticsTemplateModal form").validate(); $("#createGroupDataStatisticsTemplateModal form").valid();
    });
    $('#createFieldDataStatisticsTemplateModal').on('shown.bs.modal', function () {
        $("#updateRootDataStatisticsTemplateModal .crete-close-btn").show();
        $("#updateRootDataStatisticsTemplateModal .update-close-btn").hide();
        var selected=dataStatisticsTemplate.tree.getSelectedNodes();
        $('#createFieldDataStatisticsTemplateModal input[name="parentId"]').val(selected.nodes[0].uuid);
        $("#createFieldDataStatisticsTemplateModal form").validate(); $("#createFieldDataStatisticsTemplateModal form").valid();
    });
    $('#updateGroupDataStatisticsTemplateModal').on('shown.bs.modal', function () {
        $("#updateRootDataStatisticsTemplateModal .crete-close-btn").hide();
        $("#updateRootDataStatisticsTemplateModal .update-close-btn").removeClass("hide").show();
        var selected=dataStatisticsTemplate.tree.getSelectedNodes();
        var nodeData=selected.nodes[0];
        $.get(dataStatisticsTemplate.getUrl+nodeData.uuid,function(data){
            data=data.data;
            $('#updateGroupDataStatisticsTemplateModal input[name="parentId"]').val(data.parentId);
            $('#updateGroupDataStatisticsTemplateModal input[name="uuid"]').val(data.uuid);
            $('#updateGroupDataStatisticsTemplateModal input[name="name"]').val(data.name);
            $('#updateGroupDataStatisticsTemplateModal input[name="colspan"]').val(data.colspan);
        });
        $("#updateGroupDataStatisticsTemplateModal form").validate(); $("#updateGroupDataStatisticsTemplateModal form").valid();
    });
    $('#updateFieldDataStatisticsTemplateModal').on('shown.bs.modal', function () {
        $("#updateRootDataStatisticsTemplateModal .crete-close-btn").hide();
        $("#updateRootDataStatisticsTemplateModal .update-close-btn").removeClass("hide").show();
        var selected=dataStatisticsTemplate.tree.getSelectedNodes();
        var nodeData=selected.nodes[0];
        $.get(dataStatisticsTemplate.getUrl+nodeData.uuid,function(data){
            data=data.data;
            $('#updateFieldDataStatisticsTemplateModal input[name="parentId"]').val(data.parentId);
            $('#updateFieldDataStatisticsTemplateModal input[name="uuid"]').val(data.uuid);
            $('#updateFieldDataStatisticsTemplateModal input[name="name"]').val(data.name);
            $('#updateFieldDataStatisticsTemplateModal input[name="colspan"]').val(data.colspan);
            $('#updateFieldDataStatisticsTemplateModal input[name="unit"]').val(data.unit);
            $('#updateFieldDataStatisticsTemplateModal input[name="decimalDigits"]').val(data.decimalDigits);
            $('#updateFieldDataStatisticsTemplateModal input[name="maxNumber"]').val(data.maxNumber);
            $('#updateFieldDataStatisticsTemplateModal input[name="minNumber"]').val(data.minNumber);
            $('#updateFieldDataStatisticsTemplateModal input[name="beyondRemind"]').val(data.beyondRemind);
        });
        $("#updateFieldDataStatisticsTemplateModal form").validate(); $("#updateFieldDataStatisticsTemplateModal form").valid();
    });
    $('#updateRootDataStatisticsTemplateModal').on('shown.bs.modal', function () {
        $("#updateRootDataStatisticsTemplateModal .crete-close-btn").hide();
        $("#updateRootDataStatisticsTemplateModal .update-close-btn").removeClass("hide").show();
        var selected=dataStatisticsTemplate.tree.getSelectedNodes();
        var nodeData=selected.nodes[0];
        $.get(dataStatisticsTemplate.getUrl+nodeData.uuid,function(data){
            data=data.data;
            $('#updateRootDataStatisticsTemplateModal input[name="uuid"]').val(data.uuid);
            $('#updateRootDataStatisticsTemplateModal input[name="name"]').val(data.name);
            $('#updateRootDataStatisticsTemplateModal input[name="colspan"]').val(data.colspan);
        });
        $("#updateRootDataStatisticsTemplateModal form").validate(); $("#updateRootDataStatisticsTemplateModal form").valid();
    });
}
dataStatisticsTemplate.openForm=function(flag){
    if(flag){
        if(dataStatisticsTemplate[flag]!=dataStatisticsTemplate.createRootModalId){
            if(dataStatisticsTemplate.tree==null){
                LobiboxUtil.notify("模板树未创建！");
                dataStatisticsTemplate.openForm('createRootModalId');
                return;
            }
            var selected=dataStatisticsTemplate.tree.getSelectedNodes();
            if(selected.zTree==null){
                LobiboxUtil.notify("模板树未创建！");
                dataStatisticsTemplate.openForm('createRootModalId');
                return;
            }else if(selected.nodes==null){
                LobiboxUtil.notify("请选择父节点！");
                return;
            }
        }
        if(dataStatisticsTemplate[flag]){
            $("#"+dataStatisticsTemplate[flag]).click();
        }else{
            LobiboxUtil.notify("打开窗口无效");
        }
    }else{
        LobiboxUtil.notify("打开窗口无效");
    }
}
dataStatisticsTemplate.submitCreateForm=function(obj){
    var modalDiv=obj.parentNode.parentNode.parentNode.parentNode.parentNode;
    ajaxUtil.submit({
        form:$("#"+modalDiv.id+" form"),
        success:function(data){
                LobiboxUtil.notify("保存成功！");
                if($("#"+modalDiv.id+" input[name='uuid']").val()==""){
                    dataStatisticsTemplate.tree.addNode(data);
                }else{
                    dataStatisticsTemplate.tree.updateNode(data);
                }
                modalUtil.close("#"+modalDiv.id);
        },
        error:function(data){
            LobiboxUtil.notify("操作失败！");
        }
    });
}
dataStatisticsTemplate.openUpdateForm=function(){
    var selected=dataStatisticsTemplate.tree.getSelectedNodes();
    var data=selected.nodes;
    if(data==null){
        LobiboxUtil.notify("请选择要更新的节点！");
        return;
    }
    if(data[0].type){
        if(data[0].type=="field"){
            $("#openUpdateFieldDataStatisticsTemplateModelBtn").click();
        }else if(data[0].type=="group"){
            $("#openUpdateGroupDataStatisticsTemplateModelBtn").click();
        }else if(data[0].type=="root"){
            $("#openUpdateRootDataStatisticsTemplateModelBtn").click();
        }
    }else{
        LobiboxUtil.notify("节点类型未定义！");
    }

}
dataStatisticsTemplate.delete=function(){
    var selected=dataStatisticsTemplate.tree.getSelectedNodes();
    if(selected.nodes==null){
        LobiboxUtil.notify("请选择要删除的节点！");
        return;
    }
    var data=selected.nodes[0];
    if(data.type!="field"){
        if(typeof data.children!="undefined"&&data.children.length>0){
            LobiboxUtil.notify("请先删除子节点！");
            return ;
        }
    }
    LobiboxUtil.confirm({msg:"确定要删除吗？",fn:function() {
        $.get(dataStatisticsTemplate.deleteUrl+data.uuid+"/"+data.parentId,function(data){
            if(data.status=="success"){
                LobiboxUtil.notify("删除成功！");
                //dataStatisticsTemplateTree.removeNode(data.data);
                dataStatisticsTemplate.tree.removeNode(data.data);
            }else{
                LobiboxUtil.notify(data.message);
            }
        });
    }});
}
dataStatisticsTemplate.showNode=function(){
    var selected=dataStatisticsTemplate.tree.getSelectedNodes();
    if(selected.nodes==null){
        LobiboxUtil.notify("请选择要修改的节点！");
        return;
    }
    var data=selected.nodes[0];
    $.get(dataStatisticsTemplate.changeStateUrl+data.uuid+"/true",function(data){
        if(data.status=="success"){
            LobiboxUtil.notify("修改成功！");
        }else{
            LobiboxUtil.notify(data.message);
        }
    });
}
dataStatisticsTemplate.hideNode=function(){
    var selected=dataStatisticsTemplate.tree.getSelectedNodes();
    if(selected.nodes==null){
        LobiboxUtil.notify("请选择要修改的节点！");
        return;
    }
    var data=selected.nodes[0];
    $.get(dataStatisticsTemplate.changeStateUrl+data.uuid+"/false",function(data){
        if(data.status=="success"){
            LobiboxUtil.notify("修改成功！");
        }else{
            LobiboxUtil.notify(data.message);
        }
    });
}
dataStatisticsTemplate.moveUp=function(){
    var selected=dataStatisticsTemplate.tree.getSelectedNodes();
    if(selected.nodes==null){
        LobiboxUtil.notify("请选择要修改的节点！");
        return;
    }
    var data=selected.nodes[0];
    $.get(dataStatisticsTemplate.moveUrl+data.uuid+"/-1",function(data){
        if(data.status=="success"){
            LobiboxUtil.notify("修改成功！");
            dataStatisticsTemplateTree.moveUp(data.data);
        }else{
            LobiboxUtil.notify(data.message);
        }
    });
}
dataStatisticsTemplate.moveDown=function(){
    var selected=dataStatisticsTemplate.tree.getSelectedNodes();
    if(selected.nodes==null){
        LobiboxUtil.notify("请选择要修改的节点！");
        return;
    }
    var data=selected.nodes[0];
    $.get(dataStatisticsTemplate.moveUrl+data.uuid+"/1",function(data){
        if(data.status=="success"){
            LobiboxUtil.notify("修改成功！");
            dataStatisticsTemplateTree.moveDown(data.data);
        }else{
            LobiboxUtil.notify(data.message);
        }
    });
}

File: src/main/java/zhtt/entity/templeate/DataStatisticsTemplate.java
package zhtt.entity.templeate;

/**
 * Created by zhtt on 2016/9/14.
 */
public class DataStatisticsTemplate {

    /**
     * 数据库中作为标记为删除的字段,某些数据，因为有其他数据相关联的原因不方便直接删除，所以做标记删除
     *
     */
    public static final String DELETE_MAEK = "dismissed";

    /**
     * 节点的类型对应的值
     */
    public class Type{
        /**
         * 根节点
         */
        public static final String ROOT="root";
        /**
         * 分组节点
         */
        public static final String GROUP="group";
        /**
         * 数据项节点
         */
        public static final String FIELD="field";
        /**
         * 本机构汇总下级后的数据
         */
        public static final String org="org";
        /**
         * 本机构内部的数据
         */
        public static final String headquarters="headquarters";
    }

    /**
     * 每天需要录入的数据
     */
    public class DataKey{
        /**
         * 数据类型：本机构内部的、本机构汇总的
         */
        public static final String dataType="dataType";
        /**
         * 哪一天的数据
         */
        public static final String date="date";
        /**
         * 数据的创建时间
         */
        public static final String datetime="datetime";
        /**
         * 数据的所属机构
         */
        public static final String orgId="orgId";
        /**
         * 数据的所属机构
         */
        public static final String orgName="orgName";
        /**
         * 数据的创建机构（有可能上级为下级创建）
         */
        public static final String createOrgId="createOrgId";
        /**
         * 数据的接收机构：下级汇总上报的数据、本级内部数据
         */
        public static final String receiveOrgId="receiveOrgId";
        /**
         * 数据的统计数字
         */
        public static final String content="content";
        /**
         * 数据的上报状态
         */
        public static final String reportState="reportState";
        /**
         * 数据的下发状态（有可能上级为下级创建或修改）
         */
        public static final String issueState="issueState";
    }

    public class DocTree{
        /**
         * 本模板的主键（由于_id为ObjectId，使用起来不方便)
         */
        public static final String uuid="uuid";
        /**
         * 本模板的的所属机构（机构为根机构或分机构）
         */
        public static final String orgId="orgId";
        /**
         * node_id
         */
        public static final String node_id="node_id";
        /**
         * node_id
         */
        public static final String datetime="datetime";
        /**
         * node_id
         */
        public static final String children="children";
    }

    /**
     * 根节点的字段
     */
    public class RootKey{
        /**
         * 创建时间
         */
        public static final String datetime="datetime";
        /**
         * 用于描述节点类型
         */
        public static final String type="type";
        /**
         * 本模板的总列数
         */
        public static final String colspan="colspan";
        /**
         * 本模板的名称
         */
        public static final String name="name";
        /**
         * 本模板的的所属机构（机构为根机构或分机构）
         */
        public static final String orgId="orgId";
        /**
         * 本模板的主键（由于_id为ObjectId，使用起来不方便)
         */
        public static final String uuid="uuid";
    }

    /**
     * 分组节点的字段
     */
    public class GroupKey{
        /**
         * 创建时间
         */
        public static final String datetime="datetime";
        /**
         * 用于描述节点类型
         */
        public static final String type="type";
        /**
         * 本字段所占列数
         */
        public static final String colspan="colspan";
        /**
         * 本列的名称
         */
        public static final String name="name";
        /**
         * 本列的父节点uuid
         */
        public static final String parentId="parentId";
        /**
         * 本列的的所属机构（机构为根机构或分机构）
         */
        public static final String orgId="orgId";
        /**
         * 本列的主键（由于_id为ObjectId，使用起来不方便)
         */
        public static final String uuid="uuid";
    }

    /**
     * 数据项节点的字段
     */
    public class FieldKey{
        /**
         * 创建时间
         */
        public static final String datetime="datetime";
        /**
         * 用于描述节点类型
         */
        public static final String type="type";
        /**
         * 本字段所占列数
         */
        public static final String colspan="colspan";
        /**
         * 本列的名称
         */
        public static final String name="name";
        /**
         * 本列的父节点uuid
         */
        public static final String parentId="parentId";
        /**
         * 本列的的所属机构（机构为根机构或分机构）
         */
        public static final String orgId="orgId";
        /**
         * 本列的主键（由于_id为ObjectId，使用起来不方便)
         */
        public static final String uuid="uuid";
        /**
         * 计量单位
         */
        public static final String unit="unit";
        /**
         * 小数位数
         */
        public static final String decimalDigits="decimalDigits";
        /**
         * 最大值
         */
        public static final String maxNumber="maxNumber";
        /**
         * 最小值
         */
        public static final String minNumber="minNumber";
        /**
         * 阀值提醒
         */
        public static final String beyondRemind="beyondRemind";
    }
}
